### DISCUSSION TITLE
accessing the renderNote in the Api?

### DISCUSSION THREAD

**Discussion by ShivaFang** [TriliumNext/Trilium#4311] [Q&A] [answered]
Hello, I'm trying to get RenderNotes working, and I have some of the basic functionality understood (and I've been able to get manually run scripts that parse through tons of notes to run code and set contents which... is slow).

However, I don't know how to access the note being rendered in order to access it's properties and hierarchy.  I'd preffer to have notes with different attributes all pointing to the same script (DRY) rather than writing a script for each renderNote (WET).

api.currentNote points to the script being run, not the note being rendered.
EDIT:  I should add I'm using the latest actual release, 0.60.4, not the beta builds.  I'm not opposed to 0.61 if there's a stable version that enhances what I'm trying to do.

**Comment by zerebos** [comment, marked as answer]
You can get the current note through `api.getActiveContextNote()` when trying to make a single render source for multiple render notes. That's what I do for [Trilium-SingleFile](https://github.com/rauenzi/Trilium-SingleFile) check here: https://github.com/rauenzi/Trilium-SingleFile/blob/main/static/renderer.js#L48

**Reply by ShivaFang** [reply]
Thanks, that did the trick.
